mailbox/api/src/main/java/org/apache/james/mailbox/MessageMoveEvent.java

```diff
@@ -69,11 +70,16 @@ public Builder messageId(Collection<MessageId> messageIds) {
             return this;
         }
 
-        public MessageMoveEvent build() {
+        public Optional<MessageMoveEvent> build() {
```

Comment:
I don't like the builder taking the decision about building the event or not. I prefer the `isNoop` thing

Reply:
I will remove this commit.